### PR TITLE
Open source Inference widgets + optimize for community contributions

### PR DESCRIPTION
**Pull request by julien-c**
**TL;DR:** demo deployment at https://60c228615382546a5cd01314--huggingface-widgets.netlify.app/

Screenshot 2021-06-10 at 17 00 37

## What's the goal of this PR?

This PR's goal is to open source our Inference widgets to let community contributors check out the code and help us build widgets for new tasks in the future 🎉

## What are non-immediate goals of this PR

- **compiling to Custom Elements** (aka "Web components"): we are using SvelteKit (https://kit.svelte.dev/) to build this project with as small a boilerplate as possible, and as of now SvelteKit does not support packaging projects containing .ts files (see https://kit.svelte.dev/docs#packaging-caveats). **No big deal IMO as SvelteKit is future-proof and will support this at some point.**
- no immediate plan to **replace our widgets in moon-landing** (the codebase that powers huggingface.co) by this exact code: the code is very easy to sync back/from (modulo a few import changes) so let's not over-optimize for this yet. For now we can just manually make sure that both codebases do not diverge.

Let me know your thoughts on the points above 💜💚

## Implementation details

- The SvelteKit app exposes a demo with all possible widgets
- you can even use it to display the widget for any model (not just a hardcoded list), given that we have an API endpoint on hf.co to fetch data from just model id
- CI/CD: you can find a URL to a Netlify deployment of the SvelteKit app in every commit touching `widgets/`

### CODE REVIEW

**Comment by wietsedv** [comment]
Great! Seems to work well, except for missing tailwind (but I assume you know that).

**Comment by wietsedv** [comment]
afbeelding

**Comment by julien-c** [comment]
## This is now ready for review 💜💚♥️💛

**Review by osanseviero — APPROVED** [review]

**Review by severo — APPROVED** [review]

File: widgets/src/lib/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	export let pipeline: string | undefined;
+</script>
+
+<div class="font-semibold flex items-center mb-2">
+	{#if !noTitle}
+		<div class="text-lg flex items-center">
+			<IconLightning classNames="-ml-1 mr-1 text-yellow-500" />
+			Hosted inference API
+		</div>
+		<a target="_blank" href="/docs">
```
Comment (severo) [inline widgets/src/lib/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte:16]: Clicking on the link button shows `Error. Probably non existent model. Error. Probably non existent model.`

Comment (julien-c) [reply widgets/src/lib/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte:16]: yep, limitation of our demo app system here (not the full hf.co env)

Comment (Pierrci) [reply widgets/src/lib/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte:16]: Even the `Error. Probably non existent model.` in double? (see line 59 of `widgets/src/routes/[...model].svelte`)

Comment (julien-c) [reply widgets/src/lib/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte:16]: @Pierrci yes no sure, I had issues locally for some reasons. Maybe a transient SvelteKit bug?
